refextract/app.py

```diff
@@ -0,0 +1,101 @@
+from refextract.utils import download_file_from_url
+
+parser = FlaskParser()
+
+
+def create_app():
+    app = Flask(__name__)
+    app.config.from_pyfile("condig.cfg", silent=True)
+
+    @app.route("/extract_journal_info", methods=["POST"])
+    @parser.use_args(
+        {
+            "publication_infos": fields.List(fields.Dict, required=True),
+            "journal_kb_data": fields.Dict(required=True),
+        },
+        locations=("json",),
+    )
+    def extract_journal_info(args):
+        publication_infos = args.pop("publication_infos")
+        journal_kb_data = args.pop("journal_kb_data")
+        extracted_publication_infos = []
+        journal_dict = {"journals": journal_kb_data}
+        try:
+            for publication_info in publication_infos:
+                extracted_publication_info = extract_journal_reference(
+                    publication_info["pubinfo_freetext"],
+                    override_kbs_files=journal_dict,
+                )
+                if not extracted_publication_info:
+                    extracted_publication_info = {}
+                extracted_publication_infos.append(extracted_publication_info)
+        except Exception:
+            return make_response(jsonify({"message": "Can not extract publiacation info data"}), 500)
+        return jsonify({"extracted_publication_infos": extracted_publication_infos})
+
+    @app.route("/extract_references_from_text", methods=["POST"])
+    @parser.use_args(
+        {
+            "text": fields.String(required=True),
+            "journal_kb_data": fields.Dict(required=True),
+        },
+        locations=("json",),
+    )
+    def extract_references_from_text(args):
+        text = args.pop("text")
+        journal_kb_data = args.pop("journal_kb_data")
+        journal_dict = {"journals": journal_kb_data}
+        try:
+            extracted_references = extract_references_from_string(
+                text,
+                override_kbs_files=journal_dict,
+                reference_format=u"{title},{volume},{page}",
+            )
+        except Exception:
+            return make_response(
+                jsonify({"message": "Can not extract references."}), 500
+            )
+        return jsonify({"extracted_references": extracted_references})
+
+    @app.route("/extract_references_from_url", methods=["POST"])
+    @parser.use_args(
+        {
+            "url": fields.String(required=True),
+            "journal_kb_data": fields.Dict(required=True),
+        },
+        locations=("json",),
+    )
+    def extract_references_from_url(args):
+        url = args.pop("url")
+        journal_kb_data = args.pop("journal_kb_data")
+        journal_dict = {"journals": journal_kb_data}
+        document_data = download_file_from_url(url)
+        tmp_file_path = "/tmp/document.pdf"
+        with open(tmp_file_path, "wb") as f:
+            f.write(document_data)
+        try:
+            extracted_references = extract_references_from_file(
```

Comment:
any reason you're not simply using `extract_references_from_url` here and reinventing it (in a worse way that will lead to funny bugs when runnning with multiple gunicorn worker processes)? https://github.com/inspirehep/refextract/blob/e974d392ab9a8f29f59265ca5576318c624a46c7/refextract/references/api.py#L53-L98

Reply:
I guess this is the code we are running for years, it's just a copy-paste

Reply:
anyway, I have created a task for next https://github.com/cern-sis/issues-inspire/issues/24